cluster/gce/util.sh

```diff
@@ -2786,18 +2786,21 @@ function create-linux-nodes() {
     this_mig_size=$((${instances_left} / (${NUM_MIGS}-${i}+1)))
     instances_left=$((instances_left-${this_mig_size}))
 
-    gcloud compute instance-groups managed \
-        create "${group_name}" \
-        --project "${PROJECT}" \
-        --zone "${ZONE}" \
-        --base-instance-name "${group_name}" \
-        --size "${this_mig_size}" \
-        --template "${template_name}" || true;
-    gcloud compute instance-groups managed wait-until-stable \
-        "${group_name}" \
-        --zone "${ZONE}" \
-        --project "${PROJECT}" \
-        --timeout "${MIG_WAIT_UNTIL_STABLE_TIMEOUT}" || true &
+    # Run instance-groups creation in parallel.
+    {
+      gcloud compute instance-groups managed \
+          create "${group_name}" \
+          --project "${PROJECT}" \
+          --zone "${ZONE}" \
+          --base-instance-name "${group_name}" \
+          --size "${this_mig_size}" \
+          --template "${template_name}" || true;
+      gcloud compute instance-groups managed wait-until-stable \
+          "${group_name}" \
+          --zone "${ZONE}" \
+          --project "${PROJECT}" \
+          --timeout "${MIG_WAIT_UNTIL_STABLE_TIMEOUT}" || true
+    } &
```

Comment:
Hmm - so what is was doing before?

@mbrannock - FYI

Reply:
Previously it was `A`; `B`& where `A` - create instance group, `B` - wait for group to become stable. The & applies only to the last command. So it was running `A`s sequentially and `B`s in background (parallel).
`A` ; `A` ; ... ; `A` || `B` || `B` ||... || `B`

I run experiment with 10 instance groups, 10 instances each. It took 330s with original script to run create-linux-nodes().  Only 62s with this new one.